sdk/program/src/account_info.rs

```diff
@@ -185,6 +185,7 @@ impl<'a> AccountInfo<'a> {
     pub fn assign(&self, new_owner: &Pubkey) {
         // Set the non-mut owner field
         unsafe {
+            #[allow(invalid_reference_casting)]
```

Comment:
@Lichtso says the compiler's being hyperbolic in calling this UB :wink:

Reply:
From Rust perspective it is UB, but in LLVM the subsequent volatile write will become pretty well defined again.